test/test_gmtime.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * Phoenix-RTOS
+ *
+ * libphoenix
+ *
+ * test/test_strftime.c
+ *
+ * Copyright 2020 Phoenix Systems
+ * Author: Marcin Brzykcy
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "time.h"
+#include "stdio.h"
+#include "stdlib.h"
+
+#define NCOLS 9
+#define T1_LEN 50
+
+static const time_t input_vector[T1_LEN];
+static const int output_vector[T1_LEN][NCOLS];
+
+void tm_to_str(struct tm *t, char *buff){
+	sprintf(buff, "tm_sec %d, tm_min %d, tm_hour %d, tm_mday %d\n"
+			"tm_mon %d tm_year %d tm_wday %d tm_yday %d tm_isdst %d",
+			t->tm_sec, t->tm_min, t->tm_hour, t->tm_mday, t->tm_mon, t->tm_year,
+			t->tm_wday, t->tm_yday, t->tm_isdst);
+
+}
+
+struct tm create_tm(const int* input){
+	struct tm t;
+	t = (struct tm) {.tm_sec=input[0], .tm_min=input[1], .tm_hour=input[2], .tm_mday=input[3], 
+					 .tm_mon=input[4], .tm_year=input[5], .tm_wday=input[6], .tm_yday=input[7], 
+					 .tm_isdst=input[8]};
+	return t;
+}
+
+int compare_tm(struct tm *t1, struct tm *t2){
+	int result = 0;
+	result+= (t1->tm_sec == t2->tm_sec) ? 0 : 1;
+	result+= (t1->tm_min == t2->tm_min) ? 0 : 1;
+	result+= (t1->tm_hour == t2->tm_hour) ? 0 : 1;
+	result+= (t1->tm_mday == t2->tm_mday) ? 0 : 1;
+	result+= (t1->tm_mon == t2->tm_mon) ? 0 : 1;
+	result+= (t1->tm_year == t2->tm_year) ? 0 : 1;
+	result+= (t1->tm_wday == t2->tm_wday) ? 0 : 1;
+	result+= (t1->tm_yday == t2->tm_yday) ? 0 : 1;
+	result+= (t1->tm_isdst == t2->tm_isdst) ? 0 : 1;
+	return result;
+}
+
+
+/* this function is used to get randomly generated input data on host */
+void generate_input_host(void){
+	int i;
+	struct tm t;
+	srand(time(NULL));
```

Comment:
Tests should be repeatable! Use some constant seed instead of time(NULL).

Reply:
This function is used to generate data on host, that can be put in test file. It is then used as constant test data. The proper test will therefore give the same output every time. It is a helper function for creating test, but I decided to leave it in case if new test data would be added in the future.